### PR TITLE
FEATURE: Add config validation pre commit hook

### PR DESCRIPTION
**Pull request by rhysh90**
Commit hook to check that a config validation file is provided and that the config file passes this validation

### CODE REVIEW

File: pre_commit/config_validation.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+module Overcommit::Hook::PreCommit
+  # Runs mocha unit test
+  class ConfigValidation < Base
+    def run
+      if File.exist?("./config-validator/index.js")
+        test_result = run_config_validation
+        return test_result.nil? ? :pass : [:fail, test_result]
+      else
+        return [:fail, "Mocha unit test directory doesn't exist"]
```
Comment (matthannah) [inline pre_commit/config_validation.rb]: Update this :+1:

**Comment by apj** [comment]
@rhysh90 Looks good (besides what Matt pointed out). Not sure if I want this in the DSTIL git-hooks, though, since it's a bit more project-specific. Although, it could potentially be useful across other projects, I'd like to have a strategy for reusable project git-hooks. Any thoughts?

**Comment by rhysh90** [comment]
@apj Could we maybe have project specific branches on this repo. Any hooks that are added during a project could be pushed there and we could add the submodule to project services using,
`git submodule add https://github.com/dstil/git-hooks#<branch-name> .git-hooks`

Then we could cherry pick any project hooks that we can use DSTIL wide and merge them into master. This might be a bit laborious to implement in the middle of a project though

**Comment by apj** [comment]
@rhysh90 That's a nice approach and idea, but I'm a little concerned about the number of branches getting unruly. Also, we'd likely have different overcommit files for different repos in the same project (would be nice to include only hooks that are needed, not superfluous hooks that are ignored because some condition is not met).

I'm happy to merge it into this repo for the time being and discuss an alternate strategy at some stage soon. We can make any adjustments from that point on.

**Comment by rhysh90** [comment]
@apj Also happy for this to be merged as is for now.

Something we could look into in the future is this fork of the icebox/git-hooks project. A feature has been added to give hooks scopes like; community, project, or user. This is done by using a specific directory structure on your repo. There is then a JSON file that is examined when downloading hooks which details what scope you repo requires or what specific hooks it requires.

If my understanding of the fork is correct that is.
  
https://github.com/git-hooks/git-hooks

**Comment by apj** [comment]
@rhysh90 Cool, I'm making a list of non-project-specific tasks, so I'll chuck a link to the associated card for it on there 👍

**Pull request merged**